games/nethack/Portfile

```diff
@@ -1,110 +1,200 @@
+                   CXX=${configure.cxx} \
+                   CPP=${configure.cpp}
 
 pre-destroot {
     file mkdir "${destroot}${prefix}/share/man/man6"
-    file mkdir "${destroot}${prefix}/share/nethackdir/save"
 }
 
-destroot.target     install manpages
-destroot.keepdirs   "${destroot}${prefix}/share/nethackdir/save/"
+destroot.target    all install manpages
+destroot.keepdirs  "${destroot}${prefix}/var/games/nethack/save/"
 
 post-destroot {
-    reinplace "s|${destroot}||" "${destroot}${prefix}/bin/nethack"
+    if {${subport} eq "nethack"} {
+        reinplace "s|${destroot}||" "${destroot}${prefix}/bin/nethack"
 
-    # Don't overwrite existing preference files
-    foreach f { logfile record } {
-        file rename ${destroot}${prefix}/share/nethackdir/${f} ${destroot}${prefix}/share/nethackdir/${f}.dist
+        # Prevent existing preferences files being overwritten.
+        foreach f { logfile record sysconf } {
+            file rename ${destroot}${prefix}/var/games/nethack/${f} \
+                ${destroot}${prefix}/var/games/nethack/${f}.dist
+        }
     }
 }
 
 post-activate {
-    # Make sure initial preference files exist
-    foreach f { logfile record } {
-        if {![file exists ${prefix}/share/nethackdir/${f}]} {
-            file copy ${prefix}/share/nethackdir/${f}.dist ${prefix}/share/nethackdir/${f}
+    if {${subport} eq "nethack"} {
+        # Make initially-missing preferences files from earlier versions.
+        foreach f { logfile record sysconf } {
+            if {![file exists ${prefix}/var/games/nethack/${f}]} {
+                file copy ${prefix}/var/games/nethack/${f}.dist \
+                    ${prefix}/var/games/nethack/${f}
+            }
         }
     }
 }
 
 variant x11 {
-    patchfiles-append \
-        x11/patch-include__config.h.diff \
-        x11/patch-sys__unix__Makefile.src.diff \
-        x11/patch-sys__unix__Makefile.top.diff
+    patchfiles-append  x11/patch-x11-interface.diff
     depends_lib-append port:xorg-libXaw
 }
 
-variant autopickup_exceptions description { Automatically pick up things onto which you move } {
-    patchfiles-append patch-include__config.h.diff
-}
+subport nethack343 {
+    name                nethack343
+    version             3.4.3
+    revision            4
+    categories          games
+    maintainers         @Nax
+    description         Classic dungeon adventure game (previous version).
+    long_description    ${description}
+    homepage            http://nethack.sourceforge.net/
+    platforms           darwin freebsd
+    license             Copyleft
+
+    master_sites        sourceforge:nethack
+    distname            nethack-343-src
+    extract.suffix      .tgz
+    checksums           ${distfiles} md5 21479c95990eefe7650df582426457f9 \
+                        ${distfiles} sha1 c26537093c38152bc0fbcec20468d975b35f59fd \
+                        ${distfiles} rmd160 42f600d24715a0b7e631b357c135761410b3ca95
```

Comment:
The checksums can remain because it is a different file. But remove the references to` ${distfiles}`.

Reply:
Yes, I wondered about that - I think the `${distfiles}` is a hangover from when the commented out variants included checksums-append.

Reply:
line 102 deleted; removed `${distfiles}` from checksums

Reply:
I am still seeing the `${distfiles}` here. Also we normally have a `sha256` checksum and NOT `md5` and `sha1`. Something like this:

```
checksums           rmd160 42f600d24715a0b7e631b357c135761410b3ca95 \
                    sha256 .... \
                    size   ...

```

Reply:
Done